src/main/java/bsh/BSHArrayInitializer.java

```diff
@@ -109,9 +110,10 @@ public Object eval( Class<?> baseType, int dimensions, CallStack callstack,
             inferType = inferCommonType(null, this, callstack, interpreter);
 
         // force MapEntry to Map output
-        if ( MapEntry.class == inferType && Void.TYPE == baseType
-                || MapEntry.class == baseType )
-            baseType = Map.class;
+        if (dimensions < 2)
```

Comment:
Why does the dimensions matter again?

Reply:
It cannot be coerced to a map if there are more than two dimensions